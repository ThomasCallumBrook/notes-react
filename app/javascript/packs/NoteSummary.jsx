import React from 'react'

class NoteSummary extends React.Component{
  handleNoteClick = () => {
    this.props.currentNote(this.props.id);
  }
  handleDeleteNote = () => {
    console.log(this.props.id);
    this.props.deleteNote(this.props.id);
  }
  render(){
    return(
      <div className ="noteSummary">
        <div className="child" onClick = {this.handleNoteClick}>
          <h3> {this.props.title.slice(0,20)}</h3>
          <p>{this.props.content.slice(0,30)}...</p>
          <p>Tags: <em>{this.props.tags.slice(0,30)}</em></p>
        </div>
        <div>
          <button className="delete-btn" onClick={this.handleDeleteNote}>&times;</button>
        </div>
      </div>
    )
  }
}
export default NoteSummary;
