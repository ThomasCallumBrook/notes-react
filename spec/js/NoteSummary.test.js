import React from 'react';
import NoteSummary from 'NoteSummary';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import sinon from 'sinon';
import toJson from 'enzyme-to-json';

const currentNote = sinon.spy();
const note = {title: "Note A", content: "What a wonderful React Monday", tags: "learning"}
test('Note Summary should render as expected', () => {
  const component = shallow(<NoteSummary {...note}/>);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
test('Note Summary should respond to a click', () => {
  const currentNote = sinon.spy();
  const component = shallow(<NoteSummary {...note} currentNote={currentNote}/>);
  component.find('.child').simulate('click', {preventDefault() {} });
  expect(currentNote.callCount).toEqual(1);
});
test('Note Summary delete button should respond to a click', () => {
  const currentNote = sinon.spy();
  const component = shallow(<NoteSummary {...note} deleteNote={currentNote}/>);
  component.find('.delete-btn').simulate('click', {preventDefault() {} });
  expect(currentNote.callCount).toEqual(1);
});
