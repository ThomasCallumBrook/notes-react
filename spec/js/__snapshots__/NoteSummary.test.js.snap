// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Summary should render as expected 1`] = `
<div
  className="noteSummary"
>
  <div
    className="child"
    onClick={[Function]}
  >
    <h3>
       
      Note A
      ...
    </h3>
    <p>
      What a wonderful React Monday
      ...
    </p>
    <p>
      Tags: 
      <em>
        learning
        ...
      </em>
    </p>
  </div>
  <div>
    <button
      className="delete-btn"
      onClick={[Function]}
    >
      Ã—
    </button>
  </div>
</div>
`;
